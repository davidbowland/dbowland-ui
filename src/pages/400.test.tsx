import ServerErrorMessage from '@components/server-error-message'
import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import { mocked } from 'jest-mock'
import React from 'react'

import BadRequest from './400'

jest.mock('@aws-amplify/analytics')
jest.mock('@components/server-error-message')

describe('400 error page', () => {
  beforeAll(() => {
    mocked(ServerErrorMessage).mockReturnValue(<></>)
  })

  it('should render ServerErrorMessage', () => {
    const expectedTitle = '400: Bad Request'
    render(<BadRequest />)
    expect(mocked(ServerErrorMessage)).toHaveBeenCalledWith(
      expect.objectContaining({ title: expectedTitle }),
      expect.anything(),
    )
    expect(mocked(ServerErrorMessage)).toHaveBeenCalledTimes(1)
  })
})
