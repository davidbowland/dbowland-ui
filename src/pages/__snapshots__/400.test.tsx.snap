// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`400 error page Ensure snapshot match 1`] = `
<main
  class="sc-dlnjwi gnIxof"
>
  <h1
    class="sc-gtsrHT elXEts"
  >
    400: Bad Request
  </h1>
  <p
    class="sc-bdnxRM fSObbe"
  >
    Your request was malformed or otherwise could not be understood by the server. Please modify your request before retrying.
    <br />
    <br />
    <a
      href="/"
    >
      Go home
    </a>
    .
  </p>
</main>
`;
